File: Swift4/Swift4/model/PartijaDvaPara.swift
//
//  PartijaDvaPara.swift
//  Swift4
//
//  Created by Tomislav Jakopec on 22/02/2020.
//  Copyright © 2020 Tomislav Jakopec. All rights reserved.
//

import Foundation
class PartijaDvaPara: Partija{
    
    public override var description: String {
        let rezultat = getRezultat();
        return "Partija DVA PARA, igra gotova: " + String(super.isIgraGotova()) + ", " +  super.getIgraci()[0].description + " i " + super.getIgraci()[1].description + ": " +
                String(rezultat.getPrvi()) +
            " | " + super.getIgraci()[2].description + " i " + super.getIgraci()[3].description + ": " + String(rezultat.getDrugi());
    }
    
}

File: JS/JS/partija/partijaDvaPara.js
import { Partija } from './partija.js'

export class PartijaDvaPara extends Partija {

    constructor(id, doKolikoSeIgra, lokacija, unosi, mjesanja, igraci) {
        super(id, doKolikoSeIgra, lokacija, unosi, mjesanja, igraci)
    }

    toString() {
        const rezultat = this.getRezultat()
        const igraci = this.getIgraci()

        return `Partija DVA PARA, igra gotova: ${this.isIgraGotova()}, ` +
            `${igraci[0]} i ${igraci[1]}: ${rezultat.getPrvi()} | ` +
            `${igraci[2]} i ${igraci[3]}: ${rezultat.getDrugi()}`
    }
}